model/infinispan/src/main/java/org/keycloak/models/cache/infinispan/RealmAdapter.java

```diff
@@ -680,8 +680,7 @@ public void removeDefaultRoles(String... defaultRoles) {
 
     @Override
     public List<ClientModel> getClients() {
-        return cacheSession.getClients(this);
-
+        return cacheSession.getClients(this, 0, cacheSession.getClientsCount(this).intValue());
```

Comment:
Realm should have an overloaded method getClients(this), that just calls getClients(this, null, null). This approach will result in extra query to get the client counts first, to then list all clients.